### PR TITLE
Remove use of deprecated googletest macro

### PR DESCRIPTION
**Pull request by s-perron**
INSTANTIATE_TEST_CASE_P has been deprecated.  We need to use
INSTANTIATE_TEST_SUITE_P instead.

### CODE REVIEW

**Comment by dneto0** [comment]
If I accept this into SPIRV-Tools then it won't play nice with the NDK build,  which uses an older version of googletest.  See https://android.googlesource.com/platform/external/googletest/+log/master   
That is updated only periodically.

**Comment by gennadiycivil** [comment]
Is there any reason not to upgrade to the newer googletest? ( Except that " it is work " :-) of course

**Comment by s-perron** [comment]
There is an issue opened to get the Andriod NDK build to move up.  We have agreement that it will happen.  Just waiting for that to happen.

**Comment by s-perron** [comment]
The Andriod NDK has been updated.  This should be good to go.

File: test/opt/unify_const_test.cpp [hunk 1]
```diff
@@ -354,8 +354,14 @@ TEST_P(UnifyFrontEndConstantParamTest, TestCase) {
   Check(expected_builder, test_builder);
 }
 
-INSTANTIATE_TEST_CASE_P(Case, UnifyFrontEndConstantParamTest,
-                        ::testing::ValuesIn(std::vector<UnifyConstantTestCase>({
+INSTANTIATE_TEST_SUITE_P(
+    Case, UnifyFrontEndConstantParamTest,
```
Comment (dneto0) [inline test/opt/unify_const_test.cpp]: This looks .... wrong.

Comment (s-perron) [reply test/opt/unify_const_test.cpp]: clang-format does not change anything.

Comment (s-perron) [reply test/opt/unify_const_test.cpp]: The format check passed for this commit, so I think I will leave it.

**Pull request merged**